koku/api/iam/view/user_preference.py

```diff
@@ -115,6 +118,8 @@ def create(self, request, *args, **kwargs):
 
         return super().create(request=request, args=args, kwargs=kwargs)
 
+    @cache_control(private=True)
+    @vary_on_headers('User-Agent', 'Cookie')
```

Comment:
I think the header we need to vary on is the `x-rh-identity` header.
https://github.com/project-koku/koku/blob/8ab793cccce66efad3a97d3dae085ec334eeba7c/koku/api/common/__init__.py#L4